src/recursive_sorting/recursive_sorting.py

```diff
@@ -1,17 +1,39 @@
-# TO-DO: complete the helpe function below to merge 2 sorted arrays
+# TO-DO: complete the help function below to merge 2 sorted arrays
+a=[3,6,9]
+b=[2,4,8]
 def merge( arrA, arrB ):
     elements = len( arrA ) + len( arrB )
-    merged_arr = [0] * elements
+    merged_arr = []
     # TO-DO
-    
-    return merged_arr
+    i = j = 0
+    while (i < len(arrA) and j < len(arrB)):
+        if arrA[i] >= arrB[j]:
+            merged_arr.append(arrB[j])
+            j+=1
+        else:
+            merged_arr.append(arrA[i])
+            i+=1
+    merged_arr += arrA[i:]
+    merged_arr += arrB[j:]
 
+    return merged_arr
+    
```

Comment:
Nice use of having index counters for each array so you don't have to use .pop(0) or something else that resizes the input arrays as you use them.

Another thing you could do is make it so merged_arr is not resizing as you add items to it either.  The starter code was hinting at this with `merged_arr = [0] * elements`  which would create it with the final size, but with each value initialized as 0.  Then you could just overwrite the 0 values at whatever index you are at instead of .appending()

this solution is great, and replacing .append() with the above is a small optimization.  you will learn more about how these array methods work behind the scenes in weeks to come.